api/all/src/main/java/io/opentelemetry/api/trace/Span.kt

```diff
@@ -0,0 +1,499 @@
+     * @param attributes the attributes that will be added; these are associated with this event, not
+     * the `Span` as for `setAttribute()`.
+     * @param timestamp the explicit event timestamp since epoch.
+     * @param unit the unit of the timestamp
+     * @return this.
+     */
+    fun addEvent(name: String?, attributes: Attributes?, timestamp: Long, unit: TimeUnit?): Span
+
+    /**
+     * Adds an event to the [Span] with the given [Attributes] and `timestamp`.
+     * Note, this `timestamp` is not the same as [System.nanoTime] but may be computed
+     * using it, for example, by taking a difference of readings from [System.nanoTime] and
+     * adding to the span start time.
+     *
+     *
+     * When possible, it is preferred to use [.addEvent] at the time the event
+     * occurred.
+     *
+     * @param name the name of the event.
+     * @param attributes the attributes that will be added; these are associated with this event, not
+     * the `Span` as for `setAttribute()`.
+     * @param timestamp the explicit event timestamp since epoch.
+     * @return this.
+     */
+    fun addEvent(name: String, attributes: Attributes, timestamp: Instant?): Span {
+        if (timestamp == null) {
+            return addEvent(name, attributes)
+        }
+        return addEvent(
+            name,
+            attributes,
+            TimeUnit.SECONDS.toNanos(timestamp.epochSecond) + timestamp.nano,
+            TimeUnit.NANOSECONDS
+        )
+    }
+
+    /**
+     * Sets the status to the `Span`.
+     *
+     *
+     * If used, this will override the default `Span` status. Default status code is [ ][StatusCode.UNSET].
+     *
+     *
+     * Only the value of the last call will be recorded, and implementations are free to ignore
+     * previous calls.
+     *
+     * @param statusCode the [StatusCode] to set.
+     * @return this.
+     */
+    fun setStatus(statusCode: StatusCode?): Span {
+        return setStatus(statusCode, "")
+    }
+
+    /**
+     * Sets the status to the `Span`.
+     *
+     *
+     * If used, this will override the default `Span` status. Default status code is [ ][StatusCode.UNSET].
+     *
+     *
+     * Only the value of the last call will be recorded, and implementations are free to ignore
+     * previous calls.
+     *
+     * @param statusCode the [StatusCode] to set.
+     * @param description the description of the `Status`.
+     * @return this.
+     */
+    fun setStatus(statusCode: StatusCode?, description: String?): Span
+
+    /**
+     * Records information about the [Throwable] to the [Span].
+     *
+     *
+     * Note that the EXCEPTION_ESCAPED value from the Semantic Conventions cannot be determined by
+     * this function. You should record this attribute manually using [ ][.recordException] if you know that an exception is escaping.
+     *
+     * @param exception the [Throwable] to record.
+     * @return this.
+     */
+    fun recordException(exception: Throwable?): Span {
```

Comment:
Consider moving to different interface as this is syntactic sugar for adding a span event (or delaying implementation for now)